compiler/rustc_mir_transform/src/coverage/spans.rs

```diff
@@ -381,6 +381,12 @@ impl<'a> CoverageSpansGenerator<'a> {
 
         let merged_prefix_len = self.curr_original_span.lo() - curr.span.lo();
         let after_macro_bang = merged_prefix_len + BytePos(visible_macro.as_str().len() as u32 + 1);
+        if self.curr().span.lo() + after_macro_bang > self.curr().span.hi() {
+            // Something is wrong with the macro name span;
+            // return now to avoid emitting malformed mappings.
+            // FIXME(#117788): Track down why this happens.
+            return;
+        }
```

Comment:
Even though `curr` is already available as a local variable, the change that extracted that variable isn't part of the current (1.74) beta branch. So instead I'm calling `self.curr()` explicitly here, so that the same lines of code can simply be copied over to beta if desired.